src/drivers/net/ionic.c

```diff
@@ -167,21 +193,36 @@ static int ionic_open(struct net_device *netdev)
 	struct ionic *ionic = netdev->priv;
 	int err;
 
-	DBG_OPROM_INFO(ionic, "\n");
+	DBG_OPROM_INFO(ionic, "MTU Size :%ld\n", netdev->mtu);
 
 	err = ionic_start_device(ionic);
 	if (err)
 		return err;
 
-	err = ionic_start_queues(ionic);
+	err = ionic_lif_rxtx_init(ionic);
 	if (err) {
-		ionic_stop_device(ionic);
-		return err;
+		DBG_OPROM_ERR(ionic, "Cannot initiate LIFs: %d for rxq,txq, aborting\n",
+                      err);
+		goto err_start_device;
 	}
 
-	ionic_check_link(netdev);
+	if (ionic_is_link_up(ionic)) {
+		err = ionic_start_queues(ionic);
+		if (err)
+			goto err_lif_rxtx_init;
+	} else {
+		DBG_OPROM_INFO(ionic, "Link is down-skipping ionic_start_queues\n");
```

Comment:
I read this wrong at first and was wondering what the work "down-skipping" meant.  Putting spaces around the hyphen would help clear this up.